fix incorrect path resolution in tidy

onur-ozkan opened the pull request:
Previously, reading the current path from the environment led to failure when invoking x from outside the source root. This change fixes this issue by passing the already resolved root path into `ui_tests::check`.

Fixes #122202

compiler-errors commented on src/tools/tidy/src/ui_tests.rs:
```diff
@@ -184,7 +185,7 @@ pub fn check(path: &Path, bless: bool, bad: &mut bool) {
 [
 "#;
-        let tidy_src = std::env::current_dir().unwrap().join("src/tools/tidy/src");
+        let tidy_src = PathBuf::from(env::args_os().nth(1).unwrap()).join("src/tools/tidy/src");
```
This solution is really unnecessarily cryptic, imo. A better solution would be to pass in the `root_path` here rather than the `tests_path`:

https://github.com/rust-lang/rust/blob/a655e648a9f94d74263108366b83e677af56e35d/src/tools/tidy/src/main.rs#L105

And then adjust the rest of this file to operate from the root, rather than the test dir.

klensy commented on src/tools/tidy/src/issues.txt:
```diff
@@ -183,8 +183,8 @@
 "ui/async-await/issue-67765-async-diagnostic.rs",
 "ui/async-await/issue-68112.rs",
-"ui/async-await/issue-68523.rs",
 "ui/async-await/issue-68523-start.rs",
+"ui/async-await/issue-68523.rs",
```
PR shouldn't affect ordering here, why it changed?

compiler-errors replied:
preexisting: these weren't sorted, see https://github.com/rust-lang/rust/pull/122161#discussion_r1517107993

workingjubilee replied:
@onur-ozkan Can you rebase this? It should remove these diffs since master should contain the relevant fix.

onur-ozkan committed:
fix incorrect path resolution in tidy

Previously, reading the current path from the environment led to failure when invoking
x from outside the source root. This change fixes this issue by passing the already
resolved root path into `ui_tests::check`.

Signed-off-by: onur-ozkan <work@onurozkan.dev>

compiler-errors commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 7c13421dc0498e05624946327b1a022f2fa7b294 has been approved by `compiler-errors`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=compiler-errors 7c13421dc0498e05624946327b1a022f2fa7b294 -->
<!-- homu: {"type":"Approved","sha":"7c13421dc0498e05624946327b1a022f2fa7b294","approver":"compiler-errors","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
